### PR TITLE
WDP-75

### PR DESCRIPTION
**Pull request by lukaszcwiklinski**
https://kindergarten.stxnext.pl/browse/WDP-75

### CODE REVIEW

File: src/presence_analyzer/templates/presence_weekday.html [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{% extends "base.html" %}
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: single quotes in jinja, python, js 
double quotes in html/css

File: src/presence_analyzer/templates/presence_weekday.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+{% block js %}
+    <script type="text/javascript">
+        (function($) {
+            $(document).ready(function(){
+                var loading = $('#loading');
+                $.getJSON("/api/v1/users", function(result) {
+                    var dropdown = $("#user_id");
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: dropdown.append($('<option />', {
       'val': this.user_id,
       'text': this.name
    });

File: src/presence_analyzer/templates/presence_weekday.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{% extends "base.html" %}
+
+{% block js %}
+    <script type="text/javascript">
+        (function($) {
+            $(document).ready(function(){
+                var loading = $('#loading');
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: enter

File: src/presence_analyzer/templates/presence_weekday.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                $.getJSON("/api/v1/users", function(result) {
+                    var dropdown = $("#user_id");
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: enter

File: src/presence_analyzer/templates/presence_weekday.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                    $.each(result, function(item) {
+                        dropdown.append($("<option />").val(this.user_id).text(this.name));
+                    });
+                    dropdown.show();
+                    loading.hide();
+                });
+                $('#user_id').change(function(){
+                    var selected_user = $("#user_id").val();
+                    var chart_div = $('#chart_div');
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: - 1 var 
- enter
- in js u use camel case
- if var is a jquery selector prefix it with $

File: src/presence_analyzer/tests.py [hunk 6]
```diff
@@ -5,6 +5,8 @@
 from __future__ import unicode_literals
 
 import datetime
+import httplib
+from flask import render_template
 import json
 import os.path
 import unittest
```
Comment (piotrgrundas) [inline src/presence_analyzer/tests.py]: sort imports...

File: src/presence_analyzer/views.py [hunk 7]
```diff
@@ -19,12 +19,11 @@
 log = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
 
-@app.route('/')
-def mainpage():
-    """
-    Redirects to front page.
-    """
-    return redirect('/static/presence_weekday.html')
+@app.route('/', defaults = {'page_name' : 'presence_weekday'})
```
Comment (piotrgrundas) [inline src/presence_analyzer/views.py]: defaults={'page_name' : 'presence_weekday'}

File: src/presence_analyzer/views.py [hunk 8]
```diff
@@ -19,12 +19,11 @@
...
+@app.route('/<string:page_name>')
+def static_page(page_name):
+    return render_template('{}.html'.format(page_name))
+
```
Comment (piotrgrundas) [inline src/presence_analyzer/views.py]: What will happen if someone passes random invalid template param? + I want to see a test for such case.

File: src/presence_analyzer/templates/presence_weekday.html (hunk 1 above)
Comment (lukaszcwiklinski) [inline src/presence_analyzer/templates/presence_weekday.html]: it's pointed for double quotes in jinja documentation and css are in single quotes currently.
http://jinja.pocoo.org/docs/2.9/templates/#child-template
http://flask.pocoo.org/docs/0.12/tutorial/templates/

File: src/presence_analyzer/views.py [hunk 9]
```diff
@@ -19,12 +19,18 @@
...
-    """
-    Redirects to front page.
-    """
-    return redirect('/static/presence_weekday.html')
+@app.route('/', defaults={'page_name' : 'presence_weekday'})
+@app.route('/<string:page_name>')
+def static_page(page_name):
+    try:
+        return render_template('{}.html'.format(page_name))
+    except abort(404) as e:
```
Comment (piotrgrundas) [inline src/presence_analyzer/views.py]: Catch appropriate exception (debug which is thrown) and just abort or render 404.html

File: src/presence_analyzer/templates/presence_weekday.html [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+                            $chartDiv.show();
+                            $loading.hide();
+                            chart.draw(data, options);
+                        });
+                    }
+                });
+            });
+        })(jQuery);
+    </script>
+
```
Comment (piotrgrundas) [inline src/presence_analyzer/templates/presence_weekday.html]: remove

**Pull request merged**